src/Components/Search/NewSearch/Mobile/SearchResultsList.tsx

```diff
@@ -0,0 +1,247 @@
+import { FC, useState } from "react"
+import {
+  RelayPaginationProp,
+  createPaginationContainer,
+  graphql,
+} from "react-relay"
+import { SearchResultsList_viewer$data } from "__generated__/SearchResultsList_viewer.graphql"
+import { extractNodes } from "Utils/extractNodes"
+import {
+  NewSuggestionItem,
+  SuggionItemOptionProps,
+} from "Components/Search/NewSearch/SuggestionItem/NewSuggestionItem"
+import { Flex, Spinner } from "@artsy/palette"
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { SearchResultsListQuery } from "__generated__/SearchResultsListQuery.graphql"
+import { useTracking } from "react-tracking"
+import * as DeprecatedSchema from "@artsy/cohesion/dist/DeprecatedSchema"
+import {
+  SearchNodeOption,
+  formatOptions,
+} from "Components/Search/NewSearch/utils/formatOptions"
+import { SeeFullResults } from "Components/Search/NewSearch/Mobile/SearchResultsList/SeeFullResults"
+import { InfiniteScrollSentinel } from "Components/Search/NewSearch/Mobile/SearchResultsList/InfiniteScrollSentinel"
+import { ContentPlaceholder } from "Components/Search/NewSearch/Mobile/SearchResultsList/ContentPlaceholder"
+import { NoResults } from "Components/Search/NewSearch/Mobile/SearchResultsList/NoResults"
+
+interface SearchResultsListProps {
+  relay: RelayPaginationProp
+  viewer: SearchResultsList_viewer$data
+  query: string
+  onClose: () => void
+}
+
+const ENTITIES_PER_SCROLL = 10
+
+const SearchResultsList: FC<SearchResultsListProps> = ({
+  relay,
+  viewer,
+  query,
+  onClose,
+}) => {
+  const tracking = useTracking()
+  const [isLoading, setIsLoading] = useState(false)
+
+  const options = extractNodes(viewer.searchConnection)
+
+  if (!relay.isLoading() && options.length === 0) {
+    return <NoResults query={query} mt={50} mx={20} />
+  }
+
+  tracking.trackEvent({
+    action_type:
+      options.length > 0
+        ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
+        : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
+    query: query,
+  })
+
+  const formattedOptions: SuggionItemOptionProps[] = formatOptions(
+    options as SearchNodeOption[]
+  )
+
+  const handleLoadMore = () => {
+    if (!relay.hasMore() || relay.isLoading()) {
+      return
+    }
+
+    setIsLoading(true)
+
+    relay.loadMore(ENTITIES_PER_SCROLL, err => {
+      if (err) {
+        console.error(err)
+      }
+
+      setIsLoading(false)
+    })
+  }
+
+  const handleOnRedirect = () => {
+    onClose()
+  }
+
+  return (
+    <>
+      {formattedOptions.map((option, index) => {
+        return (
+          <NewSuggestionItem
+            query={query}
+            option={option}
+            onRedirect={handleOnRedirect}
+            key={index}
+          />
+        )
+      })}
+
+      {relay.hasMore() && <InfiniteScrollSentinel onNext={handleLoadMore} />}
+
+      {!relay.hasMore() && (
+        <SeeFullResults query={query} onClick={handleOnRedirect} />
```

Comment:
**question**: It seems to me a bit illogical to show the "See Full results" button when we are already displaying all the results (since `hasMore()` is false)

Reply:
I think you are right, I will remove "See full results" link like said [here](https://github.com/artsy/force/pull/12410#discussion_r1196179439)